pandas/plotting/_matplotlib/core.py

```diff
@@ -220,6 +224,98 @@ def __init__(
 
         self._validate_color_args()
 
+    def _validate_subplots_kwarg(
+        self, subplots: bool | Sequence[Sequence[str]]
+    ) -> bool | list[tuple[int, ...]]:
+        """
+        Validate the subplots parameter
+
+        - check type and content
+        - check for duplicate columns
+        - check for invalid column names
+        - convert column names into indices
+        - add missing columns in a group of their own
+        See comments in code below for more details.
+
+        Parameters
+        ----------
+        subplots : subplots parameters as passed to PlotAccessor
+
+        Returns
+        -------
+        validated subplots : a bool or a list of tuples of column indices. Columns
+        in the same tuple will be grouped together in the resulting plot.
+        """
+
+        if isinstance(subplots, bool):
+            return subplots
+        elif not isinstance(subplots, Iterable):
+            raise ValueError("subplots should be a bool or an iterable")
+
+        supported_kinds = (
+            "line",
+            "bar",
+            "barh",
+            "hist",
+            "kde",
+            "density",
+            "area",
+            "pie",
+        )
+        if self._kind not in supported_kinds:
+            raise ValueError(
+                "When subplots is an iterable, kind must be "
```

Comment:
What if ``subplots`` is boolean, then would it be necessary to validate ``self._kind``?
If so, then, I guess, it deserves a separate method to be invoked at the moment of ``self.kind`` initialization.

By the way, this is quite confusing for me (probably regardless of the given PR): inside ``MPLPlot`` one initializes ``self.kind``, but it seems that it is not used anywhere, only ``self._kind`` does.
Am I missing something?

Reply:
> What if subplots is boolean, then would it be necessary to validate self._kind?

When subplots is boolean we return immediately as we fall back to the current behaviour, as in master. I haven't double checked but I would hope that `self._kind` is validated separately - I mean, it's `_kind`, not `kind`.

The validation happening in `_validate_subplots_kwarg` is only for `subplots`, and its interactions with other parameters (such as `_kind`) when `subplots` is an iterable. But its job is not to validate `_kind`.